analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotDecreaseMemberVisibility.cs

```diff
@@ -45,7 +46,8 @@ public sealed class DoNotDecreaseMemberVisibility : SonarDiagnosticAnalyzer
                     var classDeclaration = (TypeDeclarationSyntax)c.Node;
 
                     if (classDeclaration.Identifier.IsMissing
-                        || !(c.ContainingSymbol is ITypeSymbol {Kind: SymbolKind.NamedType, IsSealed: false} classSymbol))
+                        || c.IsRedundantPositionalRecordContext()
```

Comment:
This class is ugly all the way from top to bottom (`ForEach` should be `foreach(...)`, emty lines, null checks, some single-use renamings in variables, condition positivity, control flow). Not sure if it's worth fixing in this PR.

Reply:
I did an additional minimal cleanup, the other parts I would leave for future work on the rule.